### PR TITLE
extensive expansion of MediaConvert adapter docs

### PR DESCRIPTION
**Pull request by jrochkind**
I had some trouble figuring out how to use the adapter, so I thought I'd write some extensive documentation to try save other people the trouble with better explanation of what's what, and also flesh out the API docs. 

Some of the behavior seems to me like maybe it's developer ergonomics could be improved, but this PR just tries to doc what's there right now. 

@mbklein @cjcolvar I think it would be awesome if you both gave this a once-over to make sure we're all on the same page.

### CODE REVIEW

File: guides/media_convert_adapter.md
Comment (cjcolvar) [inline guides/media_convert_adapter.md:63]:
```suggestion
* `log_group`. Optional, unusual. Specify the name of the CloudWatch log group to use for logging. By default, will be constructed automatically from the MediaConvert queue to use.
```

File: guides/media_convert_adapter.md
Comment (cjcolvar) [inline guides/media_convert_adapter.md:158]:
```suggestion
Where the `iamPassRole` resource is the role you will pass in the `role` configuration. The `logsStartQuery` and `logsGetQuery` permissions could probably additionally be limited to the specific CloudWatch log group.
```

**Review by cjcolvar — COMMENTED** [review]
Overall this looks good to me.  Thanks for the big helpful guide for this adapter!  I had a couple small suggestions, but I defer to @mbklein for approval since I don't really know this adapter well.

**Review by mbklein — APPROVED** [review]
This all looks great. I have nothing to quibble with. 👍🏻

**Comment by jrochkind** [comment]
Thank you for your reviews! I took your copyedits @cjcolvar and will merge when CI is green.

**Comment by jrochkind** [comment]
Oh, one question for the future, @cjcolvar , is whether we want to handle these "not actual dependency dependencies" in a more formal way.  Actually I'll go and make another ticket on that...

**Review by cjcolvar — APPROVED** [review]

**Pull request merged**
